include/exadg/operators/integrator_flags.h

```diff
@@ -22,54 +22,66 @@
 #ifndef INCLUDE_EXADG_OPERATORS_INTEGRATOR_FLAGS_H_
 #define INCLUDE_EXADG_OPERATORS_INTEGRATOR_FLAGS_H_
 
+#include <deal.II/matrix_free/evaluation_flags.h>
+
 namespace ExaDG
 {
 struct CellFlags
 {
-  CellFlags(bool const value = false, bool const gradient = false, const bool hessian = false)
-    : value(value), gradient(gradient), hessian(hessian){};
+  explicit CellFlags(
+    dealii::EvaluationFlags::EvaluationFlags const flags_in = dealii::EvaluationFlags::nothing)
+    : flags(flags_in){};
 
   CellFlags
   operator||(CellFlags const & other)
   {
     CellFlags cell_flags_combined;
 
-    cell_flags_combined.value    = this->value || other.value;
-    cell_flags_combined.gradient = this->gradient || other.gradient;
-    cell_flags_combined.hessian  = this->hessian || other.hessian;
+    cell_flags_combined.flags = flags | other.flags;
 
     return cell_flags_combined;
   }
 
-  bool value;
-  bool gradient;
-  bool hessian;
+  // checks weather any of the provided flags is set
+  inline DEAL_II_ALWAYS_INLINE //
+    bool
+    do_eval(dealii::EvaluationFlags::EvaluationFlags other =
+              dealii::EvaluationFlags::values | dealii::EvaluationFlags::gradients |
+              dealii::EvaluationFlags::hessians) const
+  {
+    return other & flags;
+  }
+
+  dealii::EvaluationFlags::EvaluationFlags flags;
 };
 
 struct FaceFlags
 {
-  FaceFlags(bool const value = false, bool const gradient = false)
-    : value(value), gradient(gradient){};
+  explicit FaceFlags(
+    dealii::EvaluationFlags::EvaluationFlags const flags_in = dealii::EvaluationFlags::nothing)
+    : flags(flags_in){};
 
   FaceFlags
   operator||(FaceFlags const & other)
```

Comment:
similar to comment above

Reply:
The default argument here is necessary to be able to call FaceFlags face_flags instad of FaceFlags face_flags{nothing}. I could add a second constructor instead if you prefer that.